File: Vectors , ArrayList, List/Java/SingleElement[Leet_Code_136].java
// Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
// You must implement a solution with a linear runtime complexity and use only constant extra space.

import java.util.ArrayList;

public class SingleElement {
    public static int singleNumber(ArrayList<Integer> nums) {
        int unique = 0;
        for (int num : nums) {
            unique ^= num;
        }
        return unique;
    }

    public static void main(String[] args) {
        ArrayList<Integer> nums = new ArrayList<>();
        nums.add(4);
        nums.add(1);
        nums.add(2);
        nums.add(1);
        nums.add(2);

        System.out.println(singleNumber(nums));
    }
}

File: Vectors , ArrayList, List/C/SingleElement[Leet_Code_136].c
// Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
// You must implement a solution with a linear runtime complexity and use only constant extra space.

#include <stdio.h>

int singleNumber(int* nums, int size) {
    int unique = 0;
    for (int i = 0; i < size; i++) {
        unique ^= nums[i];
    }
    return unique;
}

int main() {
    int nums[] = {4, 1, 2, 1, 2};
    int size = sizeof(nums) / sizeof(nums[0]);
    printf("%d\n", singleNumber(nums, size));
    return 0;
}